jac-cloud/scripts/jac-cloud.yml

```diff
@@ -0,0 +1,125 @@
+# 1. Namespace: littlex
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: littlex
+---
+# 2. ServiceAccount: jac-cloud-sa (in the littlex namespace)
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: jac-cloud-sa
+  namespace: littlex
+---
+# 3. ClusterRole: Define permissions for the jac-cloud system
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: jac-cloud-cluster-role
+rules:
+  # Core Kubernetes resources permissions
+  - apiGroups: [""]
+    resources:
+      - "namespaces"
+      - "pods"
+      - "services"
+      - "configmaps"
+    verbs: ["get", "list", "create", "delete", "watch", "patch", "update"]
+
+  # ServiceAccount permissions
+  - apiGroups: [""]
+    resources: ["serviceaccounts"]
+    verbs: ["get", "list", "create", "delete", "watch", "patch", "update"]
+
+  # RBAC permissions
+  - apiGroups: ["rbac.authorization.k8s.io"]
+    resources: ["roles", "rolebindings"]
+    verbs: ["get", "list", "create", "delete", "watch", "patch", "update"]
+
+  # Apps API group for managing deployments
+  - apiGroups: ["apps"]
+    resources: ["deployments"]
+    verbs: ["get", "list", "create", "delete", "watch", "patch", "update"]
+
+  # Allow access to secrets (e.g., for OpenAI API key)
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "list", "watch"]
+---
+# 4. ClusterRoleBinding: Bind ClusterRole to the ServiceAccount
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jac-cloud-cluster-role-binding
+subjects:
+  - kind: ServiceAccount
+    name: jac-cloud-sa
+    namespace: littlex
+roleRef:
+  kind: ClusterRole
+  name: jac-cloud-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+# 5. Deployment: jac-cloud application deployment
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: jac-cloud
+  namespace: littlex
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: jac-cloud
+  template:
+    metadata:
+      labels:
+        app: jac-cloud
+    spec:
+      # Attach the ServiceAccount for RBAC
+      serviceAccountName: jac-cloud-sa
+      containers:
+      - name: jac-cloud
+        image: ashishmahendra/littlex:0.1
```

Comment:
Do we have to use a custom image? can we make this work with the python base image?